source/_patterns/01-elements/_form-elements.scss

```diff
@@ -94,14 +94,12 @@
 // TODO: maybe we could totally restructure this, work with a :not(select) instead, differentiate in between user-invalid and provide a fallback (https://github.com/db-ui/mono/pull/893), etc.
 %form-element-validation {
 	&:not([aria-invalid]) {
-		&:not(:placeholder-shown) {
-			&:valid {
-				--formElement---borderColor: #{$db-color-green-600};
-			}
-
-			&:invalid {
-				--formElement---borderColor: #{$db-color-red-500};
-			}
+		&:user-valid {
```

Comment:
In this case a non required `input` would be green. We should set `:user-valid` only if we set `:required`, `minlength`, etc. Based on the form-element you need to change this. See [this](https://github.com/db-ui/mono/blob/main/packages/components/src/styles/_form-components.scss#L209-L222).

Reply:
> In this case a non required `input` would be green. We should set `:user-valid` only if we set `:required`, `minlength`, etc. Based on the form-element you need to change this. See [this](https://github.com/db-ui/mono/blob/main/packages/components/src/styles/_form-components.scss#L209-L222).

agreed – with this change I actually wanted to just make the switch to `:user-*` and proceed with including the other parts regarding validation afterwards in another PR.